bliss/simulator/prior.py

```diff
@@ -148,53 +206,33 @@ def _sample_n_galaxies_and_stars(self, n_sources):
 
         return galaxy_bools, star_bools
 
-    def _sample_star_fluxes(self):
+    def _sample_star_fluxes(self, star_ratios):
         latent_dims = (self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources)
         fluxes = torch.zeros(
             self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources, self.n_bands
         )
-        # draw value for g-band from pareto for each light source
+        # sample value for first band from pareto for each light source
         fluxes[:, :, :, :, 0] = self._draw_truncated_pareto(
             self.star_flux_alpha, self.star_flux_min, self.star_flux_max, latent_dims
         )
-        # Compute r-band via g-band value and ratio from SDSS catalog
-        fluxes[:, :, :, :, 1:] = self.multiflux(6, fluxes[:, :, :, :, 0])
+        # Compute remaining band fluxes via ratio from SDSS catalog and sampled values
+        if self.n_bands > 1:
```

Comment:
Condition checks if the number of bands is greater than 1 because if only single-band inference is requested, there is no need for any sampling from the SDSS catalog. Also, we pass `multiflux` the `star_ratios` as computed above which refers to the particular RCFs which will be sampled to render the stars for each image in the batch.